homeassistant/components/enocean/binary_sensor.py

```diff
@@ -61,53 +65,75 @@ def device_class(self):
         return self._device_class
 
+    @property
+    def device_type(self):
+        """Return the type of this sensor."""
+        return self._device_type
+
     def value_changed(self, packet):
         """Fire an event with the data that have changed.
-
         This method is called when there is an incoming packet associated
         with this platform.
-
         Example packet data:
         - 2nd button pressed
             ['0xf6', '0x10', '0x00', '0x2d', '0xcf', '0x45', '0x30']
         - button released
             ['0xf6', '0x00', '0x00', '0x2d', '0xcf', '0x45', '0x20']
         """
-        # Energy Bow
-        pushed = None
-
-        if packet.data[6] == 0x30:
-            pushed = 1
-        elif packet.data[6] == 0x20:
-            pushed = 0
-
-        self.schedule_update_ha_state()
-
-        action = packet.data[1]
-        if action == 0x70:
-            self.which = 0
-            self.onoff = 0
-        elif action == 0x50:
-            self.which = 0
-            self.onoff = 1
-        elif action == 0x30:
-            self.which = 1
-            self.onoff = 0
-        elif action == 0x10:
-            self.which = 1
-            self.onoff = 1
-        elif action == 0x37:
-            self.which = 10
-            self.onoff = 0
-        elif action == 0x15:
-            self.which = 10
-            self.onoff = 1
-        self.hass.bus.fire(
-            EVENT_BUTTON_PRESSED,
-            {
-                "id": self.dev_id,
-                "pushed": pushed,
-                "which": self.which,
-                "onoff": self.onoff,
-            },
-        )
+
+        if packet.data[0] == 0xf6:
+          # Energy Bow
+          pushed = None
+          pressed = None
+          first_action = None
+          second_action = None
+          second_action_valid = None
+
+          
+          if self._device_type == 2:
+            packet.parse_eep(0x02, 0x02)
+            pressed = packet.parsed['EB']['raw_value']
+            first_action = packet.parsed['R1']['raw_value']
+            second_action = packet.parsed['R2']['raw_value']
+            second_action_valid = packet.parsed['SA']['raw_value']
+
+          if packet.data[6] == 0x30:
+              pushed = 1
+          elif packet.data[6] == 0x20:
+              pushed = 0
+
+          self.schedule_update_ha_state()
+
+          action = packet.data[1]
+          if action == 0x70:
+              self.which = 0
+              self.onoff = 0
+          elif action == 0x50:
+              self.which = 0
+              self.onoff = 1
+          elif action == 0x30:
+              self.which = 1
+              self.onoff = 0
+          elif action == 0x10:
+              self.which = 1
+              self.onoff = 1
+          elif action == 0x37:
+              self.which = 10
+              self.onoff = 0
+          elif action == 0x15:
+              self.which = 10
+              self.onoff = 1
+
+          self.hass.bus.fire(
```

Comment:
We shouldn't change the code around this until the integration has been refactored to not fire events from entities. This has been discussed in earlier PRs for this integration.

Custom events can be fired from a non platform module in the integration and it shouldn't be linked to an entity. See eg the deconz integration.

https://github.com/home-assistant/home-assistant/blob/16dba3fa8519151cff4064fcaccfb194e4bd0ba0/homeassistant/components/deconz/deconz_event.py#L56

Reply:
Don't really know about the events fired from entities. 
What I saw is a lot of hardcoding in the file, that is not present in all enocean files (others use enocean API).
That's why I used the API parsing methods that do it all in one line and avoid the if/elif code that's now in this file. Doing that, as I said I avoided breaking everything so I left the existing code and rather added new variables un the event that's fired.

With the new variables, for automation I currectly only need thses two to know which button has been pressed and not launch twice the actions (button pressed and then button released):
```
      pressed: 1
      first_action: 0
```

Reply:
Just reiterating that we will not accept changes to the event firing in this integration, until that has been changed to be in line with our design rules.

Please remove the event fire changes from this PR. A separate PR needs to be done first where the event firing is reworked.